lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -1233,10 +1239,15 @@ class ConvertTritonGPUOpToLLVMPatternBase {
     auto warpsPerCTA = mfmaLayout.getWarpsPerCTA();
 
     SmallVector<unsigned> numWarpsPerDim(2);
+    unsigned mDim = mfmaLayout.getMDim();
+    unsigned nDim = mfmaLayout.getNDim();
+    assert((mDim == nDim && (mDim == 32 || mDim == 16 || mDim == 4)) ||
```

Comment:
Do we really need this kind of assert everywhere? Can we only check once at the earliest from the codegen?

Reply:
I agree, this code is redundant most of the time. It just helps add new layouts and do not forget anything crucial.

You can set new (unsupported) m/n combination in accelerate matmul pass, run tests and see where these asserts fire.